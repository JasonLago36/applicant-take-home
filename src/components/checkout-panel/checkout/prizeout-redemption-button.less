.prizeout-redemption-button {
    --prizeout-redemption-color: #DFE2E5;
    --prizeout-redemption-color-selected: #6492FB;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 0.5rem;
    width: 5rem;
    height: 1.875rem;
    cursor: pointer;
    font-size: smaller;

    &__not-selected {
        background-color: var(--prizeout-redemption-color);
        color: black;
    }

    &__selected {
        background-color: var(--color-primary);
        color: white;
    }
    
}
