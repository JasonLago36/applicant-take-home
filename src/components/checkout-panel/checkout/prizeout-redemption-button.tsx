import React from 'react';
import ClassNames from 'classnames';
import './prizeout-redemption-button.less';
import { useAppSelector } from '../../../hooks';
import { useDispatch } from 'react-redux';
import { selectedOfferButton, selectedRedemptionSelector } from '../../../slices/chosen-offer-slice';
import PropTypes from 'prop-types';

interface PrizeOutSelectedRedemptionButtonProps {
    id?: string;
    amount?: string;
    onClick?: () => void;
}
export const PrizeOutSelectedRedemptionButton: React.FC<PrizeOutSelectedRedemptionButtonProps> = ({
    id,
    amount,
}): React.ReactElement => {
    const buttonId = id;
    const dispatch = useDispatch();
    const prizeoutRedemptionAmount = amount;
    const { checkoutId } = useAppSelector(selectedRedemptionSelector);
    const classes: string = ClassNames('prizeout-redemption-button', {
        'prizeout-redemption-button__not-selected': checkoutId !== buttonId,
        'prizeout-redemption-button__selected': checkoutId === buttonId,
    });

    return (
        <div role="button" onClick={() => dispatch(selectedOfferButton(checkoutId))} className={classes}>
            {prizeoutRedemptionAmount}
        </div>
    );
};

PrizeOutSelectedRedemptionButton.propTypes = {
    amount: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
};

export default PrizeOutSelectedRedemptionButton;
