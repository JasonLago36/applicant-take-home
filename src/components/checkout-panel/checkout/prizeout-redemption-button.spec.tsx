import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import { PrizeOutSelectedRedemptionButton } from './prizeout-redemption-button';
import { Provider } from 'react-redux';
import createStore from '../../../store';
import '@testing-library/jest-dom/extend-expect';

describe('Test PrizeOutRedemption Component', () => {
    test('Component matches snapshot', () => {
        const { asFragment } = render(
            <Provider store={createStore}>
                <PrizeOutSelectedRedemptionButton id="exampleId" amount="10" />
            </Provider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('Component Prizeout Selected renders', () => {
        render(
            <Provider store={createStore}>
                <PrizeOutSelectedRedemptionButton id="exampleId" amount="10" />
            </Provider>,
        );
        const divElement = screen.getByRole('button');
        expect(divElement).toBeInTheDocument();
    });

    test('component Prizeout Selected renders with correct class', () => {
        render(
            <Provider store={createStore}>
                <PrizeOutSelectedRedemptionButton id="exampleId" amount="10" />
            </Provider>,
        );
        const divElement = screen.getByRole('button');
        expect(divElement).toHaveClass('prizeout-redemption-button__not-selected');
    });
});
