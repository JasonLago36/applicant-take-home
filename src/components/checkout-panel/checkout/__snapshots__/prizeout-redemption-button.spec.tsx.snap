// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PrizeOutRedemption Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="prizeout-redemption-button prizeout-redemption-button__not-selected"
    role="button"
  >
    10
  </div>
</DocumentFragment>
`;
